miditoolkit/midi/containers.py

```diff
@@ -284,26 +284,18 @@ def __init__(
         self.control_changes = [] if control_changes is None else control_changes
         self.pedals = [] if pedals is None else pedals
 
-    def remove_invalid_notes(self, verbose: bool = True):
-        """Removes any notes whose end time is before or at their start time."""
-        # Crete a list of all invalid notes
-        notes_to_delete = [note for note in self.notes if note.end <= note.start]
-        if verbose:
-            if len(notes_to_delete):
-                print("\nInvalid notes:\n", notes_to_delete, "\n\n")  # noqa: T201
-            else:
-                print("no invalid notes found")  # noqa: T201
-            return True
-
-        # Remove the notes found
-        for note in notes_to_delete:
-            self.notes.remove(note)
-        return False
+    def remove_notes_with_no_duration(self):
+        """Removes (inplace) notes whose end time is before or at their start time."""
+        for i in range(self.nb_notes - 1, -1, -1):
+            if self.notes[i].start >= self.notes[i].end:
+                del self.notes[i]
+
+    @property
+    def nb_notes(self) -> int:
```

Comment:
FWIW, _nb._ for "number" is a francophone contraction – I'd use `note_count`, `num_notes` or similar so it's more understandable for English speakers.

Reply:
Indeed, ty for pointing that out! Fixed in the next commit